### DISCUSSION TITLE
How to set  $env.xxx (environment variables)  in Custom commands?

### DISCUSSION THREAD

**Discussion by zerlei** [nushell/nushell#11413] [Q&A] [answered]
1. bash  to nu
![image](https://github.com/nushell/nushell/assets/40014687/15f1f7cf-a33b-4475-a2c6-fdc6d4dcec7a)

2. I append this in my config.nu
``` nushell
def proxy_on [] {
	$env.http_proxy = http://127.0.0.1:7890
	$env.https_proxy = http://127.0.0.1:7890
	$env.no_proxy = 127.0.0.1,localhost
        $env.HTTP_PROXY = http://127.0.0.1:7890
        $env.HTTPS_PROXY = http://127.0.0.1:7890
 	$env.NO_PROXY = 127.0.0.1,localhost
        echo "http_proxy:" $env.http_proxy
         echo "proxy opened！"
}

def proxy_off [] {
	hide-env http_proxy
	hide-env https_proxy
	hide-env no_proxy
        hide-env  HTTP_PROXY
	hide-env  HTTPS_PROXY
	hide-env  NO_PROXY
        echo "proxy_closed！"
}
```
3. it didn't work...

![image](https://github.com/nushell/nushell/assets/40014687/13413431-ea72-4209-be7e-8951d0a2b72c)

**Comment by zerlei** [comment, marked as answer]
I'm sorry, I found the answer in the documentation...

![image](https://github.com/nushell/nushell/assets/40014687/be780389-39b6-4410-8c28-8452968d0d0d)

https://www.nushell.sh/book/environment.html#defining-environment-from-custom-commands

**Reply by pohart** [reply]
The answer is correct, but is now located under custom-commands in the book:
https://www.nushell.sh/book/custom_commands.html#changing-the-environment-in-a-custom-command
